tests/test_core_schedule.py

```diff
@@ -1097,3 +1162,262 @@ def test_building_trips_dataframe(schedule):
         by=['route', 'trip', 'departure_time']).reset_index(drop=True)
 
     assert_frame_equal(df.sort_index(axis=1), correct_df.sort_index(axis=1))
+
+
+def test_generating_vehicles(schedule):
+    schedule.generate_vehicles()
+    assert_semantically_equal(schedule.vehicles, {'veh_3_bus': {'type': 'bus'}, 'veh_2_bus': {'type': 'bus'},
+                                         'veh_1_bus': {'type': 'bus'}, 'veh_4_bus': {'type': 'bus'}})
+
+
+def test_generating_vehicles_with_shared_vehicles_and_consistent_modes(mocker, schedule):
+    schedule.vehicles = {}
+    mocker.patch.object(DataFrame, 'drop',
+                        return_value=DataFrame({'vehicle_id': ['v_1', 'v_2', 'v_1', 'v_2', 'v_3'],
+                                                'type': ['bus', 'bus', 'bus', 'bus', 'rail']}))
+    schedule.generate_vehicles()
+    assert_semantically_equal(schedule.vehicles, {'v_1': {'type': 'bus'}, 'v_2': {'type': 'bus'},
+                                                  'v_3': {'type': 'rail'}})
+
+
+def test_generating_additional_vehicles_by_default(schedule):
```

Comment:
Are we testing two things here? The test name suggests we're going to assert that an additional vehicle has been generated, but the assertion also seems to be checking that the modes have changed for the pre-existing vehicles. If so, these might be better in separate tests.

Reply:
nah, it's checking that all the others haven't changed, they are all the weird `_bus` type